Hazelcast.Net/Hazelcast.IO.Serialization/HazelcastJsonValue.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Hazelcast.Util;
+using System;
+
+namespace Hazelcast.IO.Serialization
+{
+    /// <summary>
+    ///  Class for serializing/deserializing Hazelcast JSON types
+    /// </summary>
+    public class HazelcastJsonValue
+    {
+        private string _jsonString;
+
+        internal HazelcastJsonValue(string json)
+        {
+            _jsonString = json;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != GetType()) return false;
+
+            return Equals((HazelcastJsonValue)obj);
+        }
+
+        public override int GetHashCode()
+        {
+            return (_jsonString != null ? _jsonString.GetHashCode() : 0);
+        }
+
+        public override string ToString()
+        {
+            return _jsonString;
+        }
+
+        protected bool Equals(HazelcastJsonValue other)
+        {
+            return string.Equals(_jsonString, other._jsonString);
+        }
+
+        public static HazelcastJsonValue FromString(string jsonString)
+        {
+            return new HazelcastJsonValue(
```

Comment:
It's better to keep the validation on a separate line.

Reply:
This `ValidationUtil.CheckNotNull()` sends you back the result if it's valid so it's better to use it in this dedicated form. On separate line it looks we use just `side effect` of this function

Reply:
Yes we'll use the side effect you don't need any return value. if it is null then it will throw an exception.
You can check the other usages of this method.

Reply:
Ok. I will send such change in a moment

Reply:
Changed